imap_processing/ultra/l1c/make_helio_index_maps.py

```diff
@@ -0,0 +1,335 @@
+        If True, compute boundary scale factors (default False).
+    boundary_points : int, optional
+        Number of boundary points to sample per pixel (default 8).
+
+    Returns
+    -------
+    xarray.Dataset
+        Dataset with dimensions (step, energy, pixel) containing index,
+        theta, phi, and bsf data variables, plus ra and dec coordinates.
+    """
+    # Get spacecraft velocity at start time
+    state = imap_state(start_et, ref_frame=SpiceFrame.IMAP_DPS, observer=SpiceBody.SUN)
+    sc_vel = state[3:6]  # Extract [vx, vy, vz]
+
+    logger.info("Spacecraft velocity: %s km/s", sc_vel)
+    logger.info("Speed: %.2f km/s", np.linalg.norm(sc_vel))
+
+    # Build energy bins
+    _, energy_midpoints, energy_bin_geometric_means = build_energy_bins()
+    num_energy_bins = len(energy_bin_geometric_means)
+
+    # Get number of pixels
+    npix = hp.nside2npix(nside)
+
+    # Compute RA/Dec for pixel centers
+    pixel_indices = np.arange(npix)
+
+    # Time parameters
+    end_et = start_et + spin_duration
+    dt_step = spin_duration / num_steps
+
+    # Pre-compute all pixel vectors once
+    pixel_vecs = np.array(hp.pix2vec(nside, pixel_indices, nest=False)).T  # (npix, 3)
+
+    # Initialize output arrays
+    index_map = np.zeros((num_steps, num_energy_bins, npix))
+    theta_map = np.zeros((num_steps, num_energy_bins, npix))
+    phi_map = np.zeros((num_steps, num_energy_bins, npix))
+    bsf_map = np.zeros((num_steps, num_energy_bins, npix))
+
+    logger.info(
+        "Processing %d time steps, %d energy bins, %d pixels...",
+        num_steps,
+        num_energy_bins,
+        npix,
+    )
+    if compute_bsf:
+        logger.info(
+            "Computing boundary scale factors with %d points per pixel",
+            boundary_points,
+        )
+    # TODO vectorize loop
+    time_id = 0
+    t = start_et
+    while t < (end_et - dt_step / 2):
+        # Get rotation matrix for this time step
+        rotation_matrix = get_rotation_matrix(
+            np.array([t]),
+            from_frame=instrument_frame,
+            to_frame=SpiceFrame.IMAP_DPS,
+        )[0]
+        for energy_id in range(num_energy_bins):
+            # Convert energy to velocity (km/s)
+            energy_mean = energy_bin_geometric_means[energy_id]
+            kps = (
+                np.sqrt(2 * energy_mean * UltraConstants.KEV_J / UltraConstants.MASS_H)
+                / 1e3
+            )
+
+            # Transform pixel vectors to heliocentric frame
+            helio_velocity = (
+                sc_vel.reshape(1, 3) + kps * pixel_vecs
+            )  # Galilean transform
+            helio_normalized = helio_velocity / np.linalg.norm(
+                helio_velocity, axis=1, keepdims=True
+            )
+
+            # Transform to inst
+            inst_vecs = helio_normalized @ rotation_matrix
+            theta, phi = vector_ijk_to_theta_phi(inst_vecs)
+
+            phi = np.where(phi > np.pi, phi - 2 * np.pi, phi)
```

Comment:
I think that this is converting to what SPICE calls latitudinal coordinates. You could use this function.  The documentation is incomplete, but if you look at the SPICE docs, I think it returns what you want.

https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/spice/geometry.py#L615
https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_docs_N0067/C/cspice/reclat_c.html
```suggestion
            latitudinal = cartesian_to_latitudinal(inst_vecs)
            theta = latitudinal [..., 2]
            phi = latitudinal[..., 1]
```